docs/tutorial.md

```diff
@@ -0,0 +1,227 @@
+
+```js
+function echo(username, message) {
+  if (username === bot.username) return;
+  bot.chat(message);
+}
+
+bot.on("chat", echo); // Only emitted when a player chats publicly.
+```
+
+`bot.on('chat', echo)` will listen for [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and execute `echo` function with arguments `username`, `message`, `translate`, `jsonMsg`, and `matches` when the event emitted.
+
+#### Welcome Bot
+
+Here we're listening for [`playerJoined`](http://mineflayer.prismarine.js.org/#/api?id=quotplayerjoinedquot-player) event.
+And sending welcome message to chat.
+
+```js
+function joined(player) {
+  bot.chat(`Welcome ${player}! :D`);
+}
+
+bot.on("playerJoined", joined);
+```
+
+### Callbacks
+
+Callback is a function as argument to the main function, when the main function finish or got an error then the callback executed.
+In [the API page](http://mineflayer.prismarine.js.org/#/api) callback abbreviated as `cb` or `callback`.
+Callback usually takes parameter called `err` as the first and the second (if specified) is a value that is passed to the callback,
+if `err` is not `null` then error must be happened in the main function.
+
+Examples :
+
+#### Craft Stick Bot
+
+Here we're crafting oak logs into sticks.
+Craft oak logs into oak wood planks, wait for crafting finish. Then craft the oak wood planks into sticks.
+
+Incorect aproach ❌ :
+
+```js
+const plankRecipe = bot.recipesFor(5); // 5 ID for Oak Wood Planks
+bot.craft(plankRecipe, 1); // ❌
+const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+bot.craft(plankRecipe, 1); // ❌
+// You don't wanna start crafting sticks, before crafting logs into planks finished.
+```
+
+Correct approach with callbacks ✔️ :
+
+```js
+const plankRecipe = bot.recipesFor(5)[0]; // 5 ID for Oak Wood Planks
+
+bot.craft(plankRecipe, 1, null, function (err) {
+  // if bot.craft(plankRecipe, ...) then the code bellow get executed ✔️
+  if (err) return bot.chat(err.message);
+  // if error happened when crafting `plankRecipe` then send to chat the `err.message`
+
+  const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+  bot.craft(stickRecipe, 1, null, function (err) {
+    if (err) return bot.chat(err.message);
+    // if error happened when crafting `stickRecipe` then send to chat the `err.message`
+
+    bot.chat("Crafting Sticks finished");
+  });
+});
+```
+
+More on [`bot.craft()`](http://mineflayer.prismarine.js.org/#/api?id=botcraftrecipe-count-craftingtable-callback) method.
+
+#### Consuming Bot
+
+In this example we create bot that eat / consume held item if hungry.
+Using
+[`health`](http://mineflayer.prismarine.js.org/#/api?id=health) event of the bot to listen on any changes in health or food points.
+[`bot.consume()`](http://mineflayer.prismarine.js.org/#/api?id=botconsumecallback) method to eat.
```

Comment:
`method that makes bot eat currently held item`

Reply:
Ok